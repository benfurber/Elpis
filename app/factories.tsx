import { Comment, Community, Reply, User } from "interfaces";

export const community: Community = {
  avatarPath: require("assets/images/empower_two_women_logo.png"),
  id: "1235we",
  name: "Empodere Duas Mulheres",
  posts: [],
};

export const user: User = {
  avatarPath: require("assets/images/profile-pic-annon.png"),
  id: "1235we",
  name: "May",
};

export const post = {
  author: community,
  comments: [],
  content: null,
  id: "10387-314fs-12asdbj",
  imagePath: require("assets/images/image_post_1.jpg"),
  publishedAt: new Date("2000-01-01"),
};

export const preComment: Comment = {
  author: user,
  content:
    "Dá certo sim, o meu pai, por exemplo, fugiu quando eu tinha 5 anos e eu não faço ideia da onde ele esteja. Nunca mais voltou, pena que a violência só piorou.",
  discussionLevel: 0,
  edited: false,
  id: "21097",
  isAuthorCurrentUser: false,
  publishedAt: new Date("2019-01-01"),
  replies: [],
  totalReplies: 0,
};

export const reply: Reply = {
  author: user,
  comment: preComment,
  content: "Nao.",
  edited: false,
  id: "21097",
  isAuthorCurrentUser: false,
  link: null,
  publishedAt: new Date("2019-01-01"),
};

export const comment: Comment = {
  author: user,
  content:
    "Dá certo sim, o meu pai, por exemplo, fugiu quando eu tinha 5 anos e eu não faço ideia da onde ele esteja. Nunca mais voltou, pena que a violência só piorou.",
  discussionLevel: 0,
  edited: false,
  id: "21097",
  isAuthorCurrentUser: false,
  publishedAt: new Date("2019-01-01"),
  replies: [reply],
  title: "Meu pai fez o que ela mandou",
  totalReplies: 1,
};
