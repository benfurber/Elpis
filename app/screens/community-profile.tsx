import React, { Component } from "react";
import { StyleSheet, View } from "react-native";

import { withMappedNavigationParams } from "react-navigation-props-mapper";

import {
  BackgroundContainer,
  ButtonClose,
  CommunityDetails,
  Loading,
} from "components";
import { Community, NavigationType } from "interfaces";
import { COMMUNITY } from "queries";
import { colours, layout } from "styles";
import { Analytics, client } from "utils";

interface Props {
  navigation: NavigationType;
  communityId: Community["id"];
}

interface State {
  community: null | Community;
  isLoading: boolean;
}

class CommunityProfileScreen extends Component<Props, State> {
  state = {
    community: null,
    isLoading: true,
  };

  async fetchUser() {
    const { communityId } = this.props;

    const query = COMMUNITY;
    const variables = { id: communityId };

    const result = await client.query({ query, variables });
    this.setState({ community: result.data.community, isLoading: false });
  }

  componentDidMount() {
    Analytics.track("Community Profile");
    this.fetchUser();
  }

  render() {
    const { navigation } = this.props;
    const { isLoading, community } = this.state;

    return (
      <BackgroundContainer header={<ButtonClose navigation={navigation} />}>
        <View style={styles.container}>
          {isLoading && <Loading blueMode />}
          {community && (
            <CommunityDetails navigation={navigation} community={community} />
          )}
        </View>
      </BackgroundContainer>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: colours.whiteTransparent,
    borderTopLeftRadius: layout.borderRadiusL,
    borderTopRightRadius: layout.borderRadiusL,
    height: "100%",
    padding: layout.spacingL,
  },
});

const wrappedCommunityProfileScreen = withMappedNavigationParams()(
  CommunityProfileScreen,
);
export {
  wrappedCommunityProfileScreen as CommunityProfileScreen,
  CommunityProfileScreen as UnwrappedCommunityProfileScreen,
};
