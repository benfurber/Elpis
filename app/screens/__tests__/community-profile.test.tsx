import "react-native";
import React from "react";
import { shallow } from "enzyme";

import { community } from "factories";

import { UnwrappedCommunityProfileScreen } from "../community-profile";

let navigation;

describe("UserProfileScreen", () => {
  it("renders correctly when loading", () => {
    const communityId = community.id;

    const component = shallow(
      <UnwrappedCommunityProfileScreen
        navigation={navigation}
        communityId={communityId}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it("renders correctly when community is set", () => {
    const communityId = community.id;

    const component = shallow(
      <UnwrappedCommunityProfileScreen
        navigation={navigation}
        communityId={communityId}
      />,
    );
    component.setState({ isLoading: false, user: community });
    expect(component).toMatchSnapshot();
  });
});
