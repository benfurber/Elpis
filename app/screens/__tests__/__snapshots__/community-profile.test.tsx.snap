// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileScreen renders correctly when community is set 1`] = `
<BackgroundContainer
  header={<ButtonClose />}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 0.85)",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "height": "100%",
        "padding": 20,
      }
    }
  />
</BackgroundContainer>
`;

exports[`UserProfileScreen renders correctly when loading 1`] = `
<BackgroundContainer
  header={<ButtonClose />}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 0.85)",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "height": "100%",
        "padding": 20,
      }
    }
  >
    <Loading
      blueMode={true}
    />
  </View>
</BackgroundContainer>
`;
